### ISSUE TITLE
Helm upgrade from 3.10 to 4.10 fails

### ISSUE THREAD

**Issue by RichardPointon** [openebs/openebs#3768]
## Description
When trying to upgrade openebs using the helm chart i get the following error:

UPGRADE FAILED: rendered manifests contain a resource that already exists. Unable to continue with update: CustomResourceDefinition "lvmnodes.local.openebs.io" in namespace "" exists and cannot be imported into the current release: invalid ownership metadata; label validation error: missing key "app.kubernetes.io/managed-by": must be set to "Helm"; annotation validation error: missing key "meta.helm.sh/release-name": must be set to "openebs"; annotation validation error: missing key "meta.helm.sh/release-namespace": must be set to "openebs"

## Steps to Reproduce
MicroK8s 1.28.12 with OpenEBS 3.10 using openebs-hostpath

helm upgrade openebs openebs/openebs -n openebs -f old-values.yaml --version 4.1.0

**Comment by RichardPointon** [comment]
it appears that adding the following to the old-values.yaml file fixes the issue:

lvm-localpv:
  crds:
    lvmLocalPv:
      enabled: false

zfs-localpv:
  crds:
    zfsLocalPv:
      enabled: false

engines:
  replicated:
    mayastor:
      enabled: false

While the CRDS settings are mentioned in the documentation it is not clear that these are required even if you haven't been using ZFS or LVM.

https://openebs.io/docs/user-guides/upgrade

"If the upgrade is from Local PV LVM or Local PV ZFS storage solution, additional helm values must be specified with the above command to prevent upgrade process conflicts. The installed CRDs in 3.x would be in conflict with the CRDs in 4.1.0 as the chart structure has changed. Hence, they must be disabled."

The misleading statement is:

"Add both of these options, if your chart has both of these enabled."

as i didn't have either enabled

**Comment by RichardPointon** [comment]
I found that the upgrade worked on a test vm which has the following install history:

REVISION	UPDATED                 	STATUS    	CHART         	APP VERSION	DESCRIPTION     
1       	Tue Apr 25 11:58:44 2023	superseded	openebs-3.3.0 	3.3.0      	Install complete
2       	Tue Jun 20 11:42:24 2023	superseded	openebs-3.6.0 	3.6.0      	Upgrade complete
3       	Wed Aug 28 11:56:04 2024	superseded	openebs-3.10.0	3.10.0     	Upgrade complete
4       	Wed Aug 28 11:58:07 2024	superseded	openebs-4.1.0 	4.1.0      	Upgrade complete
5       	Wed Aug 28 12:10:17 2024	deployed  	openebs-4.1.0 	4.1.0      	Upgrade complete

but fails on a system where the install history is as follows:

REVISION	UPDATED                 	STATUS    	CHART         	APP VERSION	DESCRIPTION     
1       	Tue Feb 15 12:01:22 2022	superseded	openebs-3.1.0 	3.1.0      	Install complete
2       	Sun May  1 19:10:25 2022	superseded	openebs-3.2.0 	3.2.0      	Upgrade complete
3       	Tue Jul 19 13:59:39 2022	superseded	openebs-3.3.0 	3.3.0      	Upgrade complete
4       	Fri May 12 15:58:43 2023	superseded	openebs-3.6.0 	3.6.0      	Upgrade complete
5       	Fri Nov  3 17:19:05 2023	superseded	openebs-3.9.0 	3.9.0      	Upgrade complete
6       	Fri Feb 23 17:37:53 2024	deployed  	openebs-3.10.0	3.10.0     	Upgrade complete

So maybe this is something left over from version 3.1.0 or 3.2.0

Adding these flags:

--set lvm-localpv.crds.lvmLocalPv.enabled=false
--set zfs-localpv.crds.zfsLocalPv.enabled=false

moved it on a bit further but then it failed with:

Error: UPGRADE FAILED: Unable to continue with update: CustomResourceDefinition \"volumesnapshotclasses.snapshot.storage.k8s.io\" in namespace \"\" exists and cannot be imported into the current release: invalid ownership metadata; label validation error: missing key \"app.kubernetes.io/managed-by\": must be set to \"Helm\"; annotation validation error: missing key \"meta.helm.sh/release-name\": must be set to \"openebs\"; annotation validation error: missing key \"meta.helm.sh/release-namespace\": must be set to \"openebs\"

**Comment by niladrih** [comment]
You're correct on your assumptions here @RichardPointon. I'll try to improve the documentation surrounding this...

In addition to the two set flags that you have mentioned, you'd have to add this one as well  `--set openebs-crds.csi.volumeSnapshots.enabled=false`. Let me know how it goes @RichardPointon.

**Comment by RichardPointon** [comment]
@niladrih adding this setting fixed the issue and I was able to upgrade. Thank you for your help

**Comment by avishnu** [comment]
Let's update the documentation with the appropriate explanation.

**Comment by avishnu** [comment]
@niladrih @balaharish7 kindly update the status on this.

**Comment by balaharish7** [comment]
This issue is documented in the [PR #506](https://github.com/openebs/website/pull/506). This PR can be closed! @avishnu
